tracer/src/Datadog.Trace.SourceGenerators/AspectsDefinitions/AspectsDefinitionsGenerator.cs

```diff
@@ -21,181 +22,226 @@
+                // AspectMethodInsertBeforeAttribute(string targetMethod, int[] paramShift, bool[] boxParam)
+                3 => $"[AspectMethodInsertBefore({arguments[0]},\"\",[{arguments[1]}],[{arguments[2]}],[None],Default,[])]",
+                _ => throw new ArgumentException($"Could not find AspectMethodInsertBeforeAttribute overload with {arguments.Length} parameters")
+            },
+            "AspectMethodInsertAfterAttribute" => arguments.Length switch
+            {
+                // AspectMethodInsertAfterAttribute(string targetMethod)
+                1 => $"[AspectMethodInsertAfter({arguments[0]},\"\",[0],[False],[None],Default,[])]",
+                // AspectMethodInsertAfterAttribute(string targetMethod, AspectType aspectType, params VulnerabilityType[] vulnerabilityTypes)
+                3 => $"[AspectMethodInsertAfter({arguments[0]},\"\",[0],[False],[None],{arguments[1]},{Check(arguments[2])})]",
+                _ => throw new ArgumentException($"Could not find AspectMethodInsertAfterAttribute overload with {arguments.Length} parameters")
+            },
+            _ => throw new Exception()
+        };
+
+        string Check(string val, string ifEmpty = "[]")
         {
-            var type = Resolve(customAttributeArgument);
-            return Enum.ToObject(type, customAttributeArgument.Value);
+            return (string.IsNullOrEmpty(val) || val == NullLiteral || val == "[]") ? ifEmpty : val;
         }
-        else if (customAttributeArgument.Kind == TypedConstantKind.Array)
+
+        string MakeSameSize(string val, string ifEmpty = "[0]", string defaultValue = "False")
         {
-            var elementType = Resolve(customAttributeArgument);
-            var values = customAttributeArgument.Values;
-            return values.Select(a => a.Value).ToArray(elementType);
-        }
+            val = Check(val, ifEmpty);
+            int count = val.Count(c => c == ',');
+            string values = string.Empty;
+            for (int x = 0; x < count + 1; x++)
+            {
+                values += defaultValue;
+                if (x < count) { values += ","; }
+            }
 
-        return null;
+            return $"{val},[{values}]";
+        }
     }
 
-    private static Type Resolve(AttributeData? attribute)
-    {
-        Type[] aspects = new Type[] { typeof(AspectClassAttribute), typeof(AspectMethodReplaceAttribute), typeof(AspectMethodInsertBeforeAttribute), typeof(AspectMethodInsertAfterAttribute), typeof(AspectCtorReplaceAttribute) };
-        var name = attribute?.AttributeClass?.Name ?? string.Empty;
-        var aspect = aspects.First(a => a.Name == name);
-        return aspect;
-    }
+    private static DiagnosticInfo CreateInfo(SyntaxNode? currentNode)
+    => new(
+        new DiagnosticDescriptor(
+            "AG1",
+            "AspectsGenerator Error",
+            "Error message",
```

Comment:
Is `"Error Message"` really the correct string 🤔

Reply:
No, I'm not issuing dignostics by now. It's a placeholder